pkg/oci/layout/image.go

```diff
@@ -0,0 +1,103 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package layout
+
+import (
+	"fmt"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/siglayer"
+)
+
+// SignedImage provides access to a remote image reference, and its signatures.
+func SignedImage(path string) (oci.SignedImage, error) {
+	p, err := layout.FromPath(imagePath(path))
+	if err != nil {
+		return nil, err
+	}
+	img, err := p.Image(v1.Hash{})
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I'm less familiar with this path, so I'll have to do some reading of the signatures in here before I can weigh in.

cc @jonjohnsonjr  @imjasonh

Reply:
I'm not sure what I'm being asked to comment on by being mentioned in this review comment. `p.Image(v1.Hash{})` looks very weird to me though.

Reply:
Yah, so if there's only a single image then `p.Image(v1.Hash{})` should work -- I ended up storing a single image at different subpaths in the directory so I think it should be ok (ref https://github.com/google/go-containerregistry/pull/710)